src/daft-local-execution/src/runtime_stats.rs

```diff
@@ -20,10 +21,114 @@ use crate::{
     progress_bar::OperatorProgressBar,
 };
 
+// ----------------------- General Traits for Runtime Stat Collection ----------------------- //
+pub(crate) trait AdditionalRuntimeStats: std::fmt::Debug {
+    fn display(
+        &self,
+        w: &mut dyn Write,
+        rows_received: u64,
+        rows_emitted: u64,
+        cpu_us: u64,
+    ) -> Result<(), fmt::Error>;
+}
```

Comment:
instead of having a custom `display` i'd prefer if this returned a key value pair so that the renderer can determine how to display it. This way, it could be easily reused for the otel and dashboard subscribers.

Reply:
Makes complete sense